src/clients/Deluge.ts

```diff
@@ -182,29 +213,38 @@ export default class Deluge implements TorrentClient {
 		try {
 			let torrentInfo: TorrentInfo;
 			const { duplicateCategories } = getRuntimeConfig();
+
 			if (searchee.infoHash) {
 				torrentInfo = await this.getTorrentInfo(searchee);
 				if (!torrentInfo.complete) {
-					if (
-						torrentInfo.save_path == "missing" &&
-						!path &&
-						!searchee.path
-					) {
-						return InjectionResult.FAILURE;
-					}
 					return InjectionResult.TORRENT_NOT_COMPLETE;
 				}
 			}
+			if (!path && (!searchee.infoHash || !torrentInfo)) {
+				logger.debug({
+					label: Label.DELUGE,
+					message: `Injection failure: ${newTorrent.name} was missing critical data.`,
+				});
+				return InjectionResult.FAILURE;
+			}
 
 			const params = this.formatData(
-				`${newTorrent.getFileSystemSafeName()}.cross-seed.torrent`,
+				`${newTorrent.name}.cross-seed.torrent`,
 				newTorrent.encode().toString("base64"),
 				path ? path : torrentInfo.save_path,
 				!!searchee.infoHash
 			);
-			const addResult = await this.call("core.add_torrent_file", params);
-
-			if (addResult?.result) {
+			const addResult = await this.call<string>(
+				"core.add_torrent_file",
+				params
+			);
+			if (!addResult) {
+				logger.debug({
+					label: Label.DELUGE,
+					message: `Injection failure: Client returned empty response.`,
+				});
+				return InjectionResult.FAILURE;
+			} else if (addResult.result) {
```

Comment:
if we're worried about deluge returning completely empty json sometimes, maybe if we put this logic into `call`, then we would get this checkfor all calls?

Is this something we _should_ be worried about?

Reply:
I think we might already account for that and this might be deprecated code, call throws on a non-JSON response which I think covers this - and I don't see this anywhere in my logs for the last few weeks.

Seems unnecessary now I think.